### PR TITLE
replace deprecated constant BLACKLIST_FOR_SERIALIZATION

### PR DESCRIPTION
**Pull request by prashant-kiwi**
Changed the constant BLACKLIST_FOR_SERIALIZATION with UNSAFE_ATTRIBUTES_FOR_SERIALIZATION.

### CODE REVIEW

**Comment by MaicolBen** [comment]
Can you fix the specs? Maybe we need to upgrade devise

**Comment by ThiagoAnunciacao** [comment]
@prashant-kiwi you need some help with the tests man?

**Comment by prashant-kiwi** [comment]
> @prashant-kiwi you need some help with the tests man?

Yes, Test is failing because I need to upgrade the devise gem and I don't know where the devise gem version is written or where I can change it. Also I am new in travis-CI so I am not able to run it on my local to run all test cases.

**Comment by prashant-kiwi** [comment]
> Can you fix the specs? Maybe we need to upgrade devise

@MaicolBen @ThiagoAnunciacao I have fixed test cases.

File: Gemfile [hunk 1]
```diff
@@ -25,6 +25,7 @@ group :development, :test do
   gem 'omniauth-apple'
   gem 'rack-cors'
   gem 'thor'
+  gem 'devise', git: 'git://github.com/heartcombo/devise.git'
```
Comment (MaicolBen) [inline Gemfile:28]: Pointing to master?

File: devise_token_auth.gemspec [hunk 2]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |s|
   s.required_ruby_version = ">= 2.2.0"
 
   s.add_dependency 'rails', '>= 4.2.0', '< 6.2'
-  s.add_dependency 'devise', '> 3.5.2', '< 5'
+  s.add_dependency 'devise', '~> 4.7', '< 5'
```
Comment (MaicolBen) [inline devise_token_auth.gemspec:26]: Deprecating all the previous devise versions?

**Review by MaicolBen — COMMENTED** [review]
I think we need to have both behaviors and check if the new variable is defined, we cannot leave a lot of people behind for this

File: devise_token_auth.gemspec (hunk 2 above)
Comment (prashant-kiwi) [reply devise_token_auth.gemspec]: code reverted

File: Gemfile (hunk 1 above)
Comment (prashant-kiwi) [reply Gemfile]: removed gem

**Review by MaicolBen — APPROVED** [review]

**Pull request merged**